pkg/nmi/server/endpointpolicy.go

```diff
@@ -136,7 +137,32 @@ func deleteEndpointPolicy(endpoint *v1.HNSEndpoint, metadataIP string) error {
 }
 
 func callHcnProxyAgent(req msg.HNSRequest) ([]byte, error) {
+	retryCount := 1
+	maxRetryCount := 4
+	var sleepFactor time.Duration = 1
+
 	klog.Info("Calling HNS Agent")
+
+	for {
+		response, err := callHcnProxyAgentInternal(req)
+		if err != nil {
+			if retryCount > maxRetryCount {
+				klog.Info("Calling HNS Agent failed after all retries, giving up")
+				return nil, err
+			}
+
+			klog.Info("Calling HNS Agent failed, will retry in %s, Error: %w", sleepFactor, err)
```

Comment:
This should be klog.Infof() of klog.Errorf()

Reply:
will retry in %s seconds

Reply:
Since the variable is duration, %s will print the second there.

Reply:
Since the variable is duration, %s will print the second there.

---
In reply to: [439687764](https://github.com/bowu24/aad-pod-identity/pull/5#discussion_r439687764) [](ancestors = 439687764)